haybaler.py

```diff
@@ -140,6 +140,59 @@ def exclude_taxa(file, path, taxa_to_exclude):
     df.to_csv(path + "/" + file, sep="\t")  # save again as csv
 
 
+def shorten_names(output_path, col, output_file, summing_short):
+    short = pd.read_csv(output_path + "/" + col + "_" + output_file, index_col = 0, sep = "\t")
+    rownames = list(short.index)
+    for row in rownames:
+        new_name = row
+        split_name = row.split("_")
+        for n in range(len(split_name)):
+            if split_name[n] == "organism" or split_name[n] == "candidatus":
+                new_name = split_name[n] + "_" + split_name[n+1] + "_" + split_name[n+2]
+                new_name = subspecies(new_name, n, 3, split_name)
+                short = check_existing(new_name, row, short, summing_short)
+                break
+            try:
+                first_letter = ord(split_name[n][0])
+                second_letter = ord(split_name[n][1])
+                second_first = ord(split_name[n+1][0])
+                if 64 < first_letter < 91 and 96 < second_letter < 123 and 96 < second_first < 123:
+                    new_name = split_name[n] + "_" + split_name[n+1]
+                    new_name = subspecies(new_name, n, 2, split_name)
+                    short = check_existing(new_name, row, short, summing_short)
+                    break
+            except:
+                pass
+
+    save_name = output_path + "/" +  col + "_" + output_file
+    save_name = save_name.split(".")[0] + "_short.csv"
+
+    index = short.index
+    if index.is_unique:  # only saved if all row names are unique
+        short.to_csv(save_name, sep='\t')
+    # if not all row names are unique: for the read count csv the values of the non unique rows are summed
+    elif col == "read_count":
+        short = short.groupby("species").sum()
+        short.to_csv(save_name, sep='\t') # save read count table
```

Comment:
Changes the row order

Maybe don't write any short csv if index is not unique?